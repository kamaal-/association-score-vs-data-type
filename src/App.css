.App {
  align-items: center;
  display: flex-start;
  justify-content: center;
  min-height: 100vh;
  padding: 40px;
  background-color: #ccc;
}

.horizontal__grids line {
  stroke: #412960;
  stroke-opacity: 0.1;
  shape-rendering: crispEdges;
}
.horizontal__grids .domain{
  display: none;
}

.tick text{
  color:  #412960;
}

.vertical__axis .domain, .horizontal__axis .domain{
  stroke: #412960;
}