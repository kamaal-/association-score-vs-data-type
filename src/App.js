import React, {useEffect, useState} from "react";
import "./App.css";
import Table from './components/DataGrid'
import {getInitialData} from './services/'

function App() {
  const [rows, setRows] = useState([]);
  const chartTitle = 'Association Score Vs Data Type'
  useEffect(() => {
    getInitialData((rows) => {
      setRows(rows)
    })
  }, [])
  return (
    <div className="App">
      <Table chartTitle={chartTitle} rows={rows}/>
    </div>
  );
}

export default App;
