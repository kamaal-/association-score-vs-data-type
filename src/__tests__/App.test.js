import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';

describe('App', () => {
  	let testable = null
  	const setup = () => {
    	return render(<App />);
  	}

  	beforeAll(() => {
    	testable = setup()
  	})

  	test('renders Symbol column', () => {
    	const symbolColumn = testable.getByText(/Symbol/i);
    	expect(symbolColumn).toBeInTheDocument();
	});
	  
	it('should match snapshot', () => {
        const { asFragment } = setup()
        expect(asFragment()).toMatchSnapshot();
    })

})
 