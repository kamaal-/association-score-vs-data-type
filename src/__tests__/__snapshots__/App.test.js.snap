// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        aria-label="caption table"
        class="MuiTable-root makeStyles-table-52"
      >
        <caption>
          Genes associated with lung carcinoma
        </caption>
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
            data-testid="title-tr"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            />
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Symbol
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Gene ID
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Gene Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              Overall Association Score
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        />
      </table>
    </div>
  </div>
</DocumentFragment>
`;
